matter/wifi/wf200/efr_spi.c

```diff
@@ -87,13 +99,10 @@ sl_status_t sl_wfx_host_init_bus(void)
   dummy_tx_data      = 0;
   usartInit.msbf     = true;
   usartInit.baudrate = 36000000u;
-#if defined(EFR32MG12)
   CMU_ClockEnable(cmuClock_HFPER, true);
-#endif
   CMU_ClockEnable(cmuClock_GPIO, true);
   CMU_ClockEnable(MY_USART_CLOCK, true);
   USART_InitSync(MY_USART, &usartInit);
```

Comment:
How does this work ? MY_USART and MY_USART_CLOCK are being define in this fine for mg24 only but here you use them for MG12

#elif defined(EFR32MG24)
#define MY_USART SL_WFX_HOST_PINOUT_SPI_PERIPHERAL

#if defined(EFR32MG24)
static uint32_t            MY_USART_CLOCK;


The way you are doing it is really confusing and to clean.

Reply:
moved these to header file and cleaned the code.